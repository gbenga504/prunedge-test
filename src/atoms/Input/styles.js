import { color, fontsize, fontWeight } from "../../Css";

export const styles = {
  label: {
    color: color.grey,
    fontSize: fontsize.base,
    fontWeight: fontWeight.regular
  },
};
