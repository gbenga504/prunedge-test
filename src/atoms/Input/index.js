import React from "react";
import PropTypes from "prop-types";
import withStyles from "@material-ui/core/styles/withStyles";
import TextField from "@material-ui/core/TextField";
import Typography from "@material-ui/core/Typography";

import { styles } from "./styles";

class Input extends React.PureComponent {
  static propTypes = {
    halfInput: PropTypes.bool,
    containerStyle: PropTypes.object,
    ...TextField.propTypes
  };

  render() {
    let {
        classes,
        children,
        defaultValue,
        halfInput,
        type,
        required,
        onChange,
        containerStyle,
        name,
        label,
        disabled,
        select,
        rowsMax,
        rows,
        multiline
      } = this.props,
      percentWidth = halfInput ? "50%" : "100%";

    return (
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          width: percentWidth,
          ...containerStyle
        }}
      >
        <Typography variant="body2" classes={{ body2: classes.label }}>
          {label}
        </Typography>
        <TextField
          type={type}
          required={required || false}
          disabled={disabled || false}
          select={select || false}
          className={classes.textField}
          defaultValue={defaultValue}
          fullWidth={true}
          multiline={multiline}
          margin="dense"
          rows={rows ? rows : ""}
          InputProps={{ style: { height: !multiline && 45 } }}
          rowsMax={rowsMax ? rowsMax : ""}
          variant="outlined"
          inputProps={{ "aria-label": "input" }}
          onChange={e => onChange(name, e.target.value)}
        >
          {children}
        </TextField>
      </div>
    );
  }
}

export default withStyles(styles)(Input);
