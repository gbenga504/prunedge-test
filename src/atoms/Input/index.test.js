import React from "react";
import { create } from "react-test-renderer";
import MenuItem from "@material-ui/core/MenuItem";

import Input from "./index";

describe("Input", () => {
  it("renders a text input with no children", () => {
    const tree = create(<Input type="text" />);
    expect(tree).toMatchSnapshot();
  });

  it("renders a select input with 2 items", () => {
    const tree = create(
      <Input
        label="Gender"
        halfInput={true}
        type="text"
        select={true}
        value=""
        name="gender"
        required={true}
      >
        <MenuItem value="male">Male</MenuItem>
        <MenuItem value="female">Female</MenuItem>
      </Input>
    );
    expect(tree).toMatchSnapshot();
  });
});
