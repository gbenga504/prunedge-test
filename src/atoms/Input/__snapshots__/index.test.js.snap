// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders a select input with 2 items 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "width": "50%",
    }
  }
>
  <p
    className="MuiTypography-root MuiTypography-body2 Input-label-1"
  >
    Gender
  </p>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      onClick={[Function]}
      style={
        Object {
          "fontSize": 14,
          "height": 45,
        }
      }
    >
      <div
        aria-haspopup="listbox"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense MuiSelect-outlined"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        aria-label="input"
        rows=""
        type="hidden"
        value=""
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-70 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-71"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`Input renders a text input with no children 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <p
    className="MuiTypography-root MuiTypography-body2 Input-label-1"
  />
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      onClick={[Function]}
      style={
        Object {
          "fontSize": 14,
          "height": 45,
        }
      }
    >
      <input
        aria-invalid={false}
        aria-label="input"
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        rows=""
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-70 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-71"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
